compiler/ast/parser.nim

```diff
@@ -1854,8 +1852,8 @@ proc parseEnum(p: var Parser): PNode =
     var symPragma = a
     if (p.tok.indent < 0 or p.tok.indent >= p.currInd) and p.tok.tokType == tkCurlyDotLe:
       let pragma = optPragmas(p)
-      symPragma = newTreeI(nkPragmaExpr, p.lineInfo, [a, pragma])
-    # nimpretty support here
+      symPragma = p.newNode(nkPragmaExpr, @[a, pragma])
+    # nimpretty support here LOL what is this even
```

Comment:
Yeah, it's likely some dumb leak of Implementation because it was "easier" to do this silly thing. 😄

Reply:
I think it would be possible o keep nimpretty in the parser for the time being, but as soon as I'm done with DOD part of the lexer it will have to go away and finally tackle the "teach a system without eyes how to make code visually appealing, how hard can it be? :P" problem.

Reply:
After some of these things design issues/boundary violations are resolved, we'll no longer be hobbled by bugs and limitations. As in no more silly compromises. That alone will do wonders for simplifying the formatting problem.